receiver/oracledbreceiver/factory.go

```diff
@@ -47,11 +50,17 @@ func createReceiverFunc(sqlOpenerFunc sqlOpenerFunc, clientProviderFunc clientPr
 		consumer consumer.Metrics,
 	) (receiver.Metrics, error) {
 		sqlCfg := cfg.(*Config)
+		sqlCfg.DataSource = getDataSource(*sqlCfg)
```

Comment:
Why do we need to mutate the original config? We only need to pass the datasource to `sqlOpenerFunc`, right?

Reply:
Yes, the only usage is to pass it to the `sqlOpenerFunc`. My thought process was that setting it here would make it less likely in the future to use the wrong variable if extra functionality was added.

Do you think it would be better to just directly call `getDataSource` later, instead of mutating the original config?

Reply:
Yes. Ideally, the original config should not be mutated

Reply:
Alright, I've removed this. To access the data source the code will just call the `getDataSource` method instead of using the config option directly.